lib/automatic/api_client.rb

```diff
@@ -0,0 +1,95 @@
+require 'automatic/api_client/trips'
+require 'automatic/api_client/vehicles'
+
+module Automatic
+  class ApiClient
```

Comment:
Seems to be a large amount of duplication here, with regards to the existing `Automatic::Client`. Have you thought about ways of sharing behavior between the two, and modifying only what is required to be different?

Seems like the main branching point is the instance-level auth token. `Automatic::Client` sets it at the class-level, and `Automatic::ApiClient` sets it at the instance-level. Otherwise, the methods seem to perform the same actions, regardless of being class or instance methods. DRYing that up would be awesome. Did you have thoughts on how you would go about that? Perhaps some kind of service model that could facilitate this?